This PR didn't change the output of nbclient, so there shouldn't be any breaking changes.

Here's the log of failures, none of them seem to be related to this change:
```
error: builder for '/nix/store/sq782czgzfi9pjrxnimvf35cs991mlq9-WolframEngine_13.0.1_LINUX.sh.drv' failed with exit code 1;
       last 9 log lines:
       >
       > ***
       > This nix expression requires that WolframEngine_13.0.1_LINUX.sh is
       > already part of the store. Download the file from
       > https://www.wolfram.com/engine/ and add it to the nix store
       > with nix-store --add-fixed sha256 <FILE>.
       >
       > ***
       >
       For full logs, run 'nix log /nix/store/sq782czgzfi9pjrxnimvf35cs991mlq9-WolframEngine_13.0.1_LINUX.sh.drv'.
error: 1 dependencies of derivation '/nix/store/3wfrc0ys9vqh46sxmsv7px7y7ybn7mvl-wolfram-engine-13.0.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/nclff1b464wiyff1cc4ddsc8mb7ws1s0-jupyter-kernels.drv' failed to build
error: builder for '/nix/store/6qiafwlgd9qdgcncf4587n4yqgzsc4ig-python3.10-chainer-7.8.1.drv' failed with exit code 4;
       last 10 log lines:
       >     import numpy.distutils.system_info
       >   File "/nix/store/jahyfd38sjr2k9q5d53q6h522rx2dzkm-python3.10-numpy-1.21.5/lib/python3.10/site-packages/numpy/distutils/__init__.py", line 24, in <module>
       >     from . import ccompiler
       >   File "/nix/store/jahyfd38sjr2k9q5d53q6h522rx2dzkm-python3.10-numpy-1.21.5/lib/python3.10/site-packages/numpy/distutils/ccompiler.py", line 8, in <module>
       >     from distutils import ccompiler
       >   File "/nix/store/j3f6fny7sa0wdf6v76016d9ic8pyw6fr-python3-3.10.4/lib/python3.10/distutils/__init__.py", line 19, in <module>
       >     warnings.warn(_DEPRECATION_MESSAGE,
       > DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
       >
       > 
       For full logs, run 'nix log /nix/store/6qiafwlgd9qdgcncf4587n4yqgzsc4ig-python3.10-chainer-7.8.1.drv'.
error: 1 dependencies of derivation '/nix/store/zp7q7qyhy023s3i5108hi6pxp8f2bghq-python3.9-notebook-6.4.10.drv' failed to build
error: 1 dependencies of derivation '/nix/store/6pqalq8215449cyj9pczf3rp5454lgrx-wolfram-notebook.drv' failed to build
error: builder for '/nix/store/x99c3hw8hh3qap45fgc3ldhllq97pf27-python3.9-phik-0.12.2.drv' failed with exit code 1;
       last 10 log lines:
       > configuring
       > no configure script, doing nothing
       > building
       > Executing setuptoolsBuildPhase
       > Traceback (most recent call last):
       >   File "/build/phik-0.12.2/nix_run_setup", line 8, in <module>
       >     exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
       >   File "setup.py", line 20, in <module>
       >     from skbuild import setup as sk_setup
       > ModuleNotFoundError: No module named 'skbuild'
       For full logs, run 'nix log /nix/store/x99c3hw8hh3qap45fgc3ldhllq97pf27-python3.9-phik-0.12.2.drv'.
error: builder for '/nix/store/7wqrpm9a9yhqrajrz3p2nk1qa5rrnv5r-python3.9-modeled-0.1.8.drv' failed with exit code 1;
       last 10 log lines:
       > adding 'modeled-0.1.8.dist-info/RECORD'
       > removing build/bdist.linux-x86_64/wheel
       > Finished executing setuptoolsBuildPhase
       > installing
       > Executing pipInstallPhase
       > /build/modeled-0.1.8/dist /build/modeled-0.1.8
       > Processing ./modeled-0.1.8-py3-none-any.whl
       > ERROR: Could not find a version that satisfies the requirement path.py>=8.0 (from modeled) (from versions: none)
       > ERROR: No matching distribution found for path.py>=8.0
       > 
       For full logs, run 'nix log /nix/store/7wqrpm9a9yhqrajrz3p2nk1qa5rrnv5r-python3.9-modeled-0.1.8.drv'.
error: 1 dependencies of derivation '/nix/store/11nadiziy7p8b7sgjzzm1zr92xj6w35l-python3.9-robotframework-tools-0.1rc4.drv' failed to build
error: builder for '/nix/store/4h01fb2xvp077yx4c11xcpi4jvcx7k7s-python3.9-pydeck-0.7.1.drv' failed with exit code 1;
       last 10 log lines:
       >     dist.parse_config_files()
       >   File "/nix/store/api34p64v2gy48lppbhffk2idmnfigv5-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/dist.py", line 854, in parse_config_files
       >     pyprojecttoml.apply_configuration(self, filename, ignore_option_errors)
       >   File "/nix/store/api34p64v2gy48lppbhffk2idmnfigv5-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/config/pyprojecttoml.py", line 55, in apply_configuration
       >     config = read_configuration(filepath, True, ignore_option_errors, dist)
       >   File "/nix/store/api34p64v2gy48lppbhffk2idmnfigv5-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/config/pyprojecttoml.py", line 121, in read_configuration
       >     validate(subset, filepath)
       >   File "/nix/store/api34p64v2gy48lppbhffk2idmnfigv5-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/config/pyprojecttoml.py", line 44, in validate
       >     raise error from None
       > ValueError: invalid pyproject.toml config: `project`
       For full logs, run 'nix log /nix/store/4h01fb2xvp077yx4c11xcpi4jvcx7k7s-python3.9-pydeck-0.7.1.drv'.
error: 1 dependencies of derivation '/nix/store/6wfk3pbb520cx1bqdwkdrpnr37gxcqsv-streamlit-1.2.0.drv' failed to build
error: builder for '/nix/store/k7fr8nw4i2jiq0ivmwrkhaipmpkgy35v-python3.10-pydeck-0.7.1.drv' failed with exit code 1;
       last 10 log lines:
       >     dist.parse_config_files()
       >   File "/nix/store/w11qj9yfy09gbdr5saa3pv60j06g3ras-python3.10-setuptools-61.2.0/lib/python3.10/site-packages/setuptools/dist.py", line 854, in parse_config_files
       >     pyprojecttoml.apply_configuration(self, filename, ignore_option_errors)
       >   File "/nix/store/w11qj9yfy09gbdr5saa3pv60j06g3ras-python3.10-setuptools-61.2.0/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py", line 55, in apply_configuration
       >     config = read_configuration(filepath, True, ignore_option_errors, dist)
       >   File "/nix/store/w11qj9yfy09gbdr5saa3pv60j06g3ras-python3.10-setuptools-61.2.0/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py", line 121, in read_configuration
       >     validate(subset, filepath)
       >   File "/nix/store/w11qj9yfy09gbdr5saa3pv60j06g3ras-python3.10-setuptools-61.2.0/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py", line 44, in validate
       >     raise error from None
       > ValueError: invalid pyproject.toml config: `project`
       For full logs, run 'nix log /nix/store/k7fr8nw4i2jiq0ivmwrkhaipmpkgy35v-python3.10-pydeck-0.7.1.drv'.
error: 1 dependencies of derivation '/nix/store/imbn975pd2axrll7mr4r2ggh5h76482l-python3.10-einops-0.4.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/2bn3w8rvafxl3wdng5ahhd5n67yg9h93-python3.10-augmax-unstable-2022-02-19.drv' failed to build
error: builder for '/nix/store/rcmy2j6nkqdd90czg2m3kd0cghn0if0i-python3.9-einops-0.4.1.drv' failed with exit code 1;
       last 10 log lines:
       > traitlets: DEBUG: msg_type: stream
       > traitlets: DEBUG: content: {'name': 'stdout', 'text': '524 ms ± 160 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)\n'}
       > traitlets: ERROR: Timeout waiting for execute reply (60s).
       > traitlets: DEBUG: Kernel is taking too long to finish, terminating
       > --------------------- >> end captured logging << ---------------------
       >
       > ----------------------------------------------------------------------
       > Ran 204 tests in 253.955s
       >
       > FAILED (errors=1)
       For full logs, run 'nix log /nix/store/rcmy2j6nkqdd90czg2m3kd0cghn0if0i-python3.9-einops-0.4.1.drv'.
error: 1 dependencies of derivation '/nix/store/gjl0csg5avm8n9z89d12rk1hgs92p3d8-python3.9-augmax-unstable-2022-02-19.drv' failed to build
error: 1 dependencies of derivation '/nix/store/6c8kvdbpifiw2qgw1wywr5p0d6fqc2px-python3.9-treex-0.6.10.drv' failed to build
error: 1 dependencies of derivation '/nix/store/dvlrizqpqbxkbqfn3683rm36m61awd6a-python3.9-elegy-0.8.6.drv' failed to build
error: builder for '/nix/store/0bkzd7p97z4hgfacjmm2qpn0w4wymxjp-python3.9-atomman-1.4.4.drv' failed with exit code 1;
       last 10 log lines:
       > =========================== short test summary info ============================
       > FAILED tests/test_unitconvert.py::Test_unitconvert::test_build_unit - Asserti...
       > FAILED tests/test_unitconvert.py::Test_unitconvert::test_set_and_get_in_units
       > FAILED tests/test_unitconvert.py::Test_unitconvert::test_set_literal - Assert...
       > FAILED tests/test_unitconvert.py::Test_unitconvert::test_scalar_model - Asser...
       > FAILED tests/tools/test_vect_angle.py::Test_vect_angle::test_0 - AssertionErr...
       > FAILED tests/tools/test_vect_angle.py::Test_vect_angle::test_180 - AssertionE...
       > FAILED tests/tools/test_vect_angle.py::Test_vect_angle::test_90 - AssertionEr...
       > FAILED tests/tools/test_vect_angle.py::Test_vect_angle::test_45 - AssertionEr...
       > =========== 8 failed, 66 passed, 4 deselected, 90 warnings in 44.75s ===========
       For full logs, run 'nix log /nix/store/0bkzd7p97z4hgfacjmm2qpn0w4wymxjp-python3.9-atomman-1.4.4.drv'.
error: builder for '/nix/store/inlxyxj5n8r952z0cbpkznsg0n44iy53-python3.10-atomman-1.4.4.drv' failed with exit code 1;
       last 10 log lines:
       > =========================== short test summary info ============================
       > FAILED tests/test_unitconvert.py::Test_unitconvert::test_build_unit - Asserti...
       > FAILED tests/test_unitconvert.py::Test_unitconvert::test_set_and_get_in_units
       > FAILED tests/test_unitconvert.py::Test_unitconvert::test_set_literal - Assert...
       > FAILED tests/test_unitconvert.py::Test_unitconvert::test_scalar_model - Asser...
       > FAILED tests/tools/test_vect_angle.py::Test_vect_angle::test_0 - AssertionErr...
       > FAILED tests/tools/test_vect_angle.py::Test_vect_angle::test_180 - AssertionE...
       > FAILED tests/tools/test_vect_angle.py::Test_vect_angle::test_90 - AssertionEr...
       > FAILED tests/tools/test_vect_angle.py::Test_vect_angle::test_45 - AssertionEr...
       > =========== 8 failed, 66 passed, 4 deselected, 91 warnings in 38.63s ===========
       For full logs, run 'nix log /nix/store/inlxyxj5n8r952z0cbpkznsg0n44iy53-python3.10-atomman-1.4.4.drv'.
error: builder for '/nix/store/v3pnzb91nlm1x6gnamh73vpzqic0f5dn-python3.10-jupyterhub-1.3.0.drv' failed with exit code 1;
       last 10 log lines:
       >   /nix/store/m1irdr1g69i09gkc9qnrx0hppnxcppz4-python3.10-pytest-7.1.1/lib/python3.10/site-packages/_pytest/runner.py:522: RuntimeWarning: coroutine 'LocalProcessSpawner.stop' was never awaited
       >     fin()
       >   Enable tracemalloc to get traceback where the object was allocated.
       >   See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
       >
       > -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
       > =========================== short test summary info ============================
       > FAILED jupyterhub/tests/test_api.py::test_spawn_handler - requests.exceptions...
       > FAILED jupyterhub/tests/test_named_servers.py::test_create_named_server - ass...
       > ==== 2 failed, 355 passed, 5 deselected, 399 warnings in 515.93s (0:08:35) =====
       For full logs, run 'nix log /nix/store/v3pnzb91nlm1x6gnamh73vpzqic0f5dn-python3.10-jupyterhub-1.3.0.drv'.
error: 1 dependencies of derivation '/nix/store/22ml30v8afklr8rw6zr0rpp2f3avli5r-python3.10-batchspawner-1.1.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/mbw1wmh528m3f5iscf4bnxclgdr7ahr3-python3.10-dockerspawner-12.1.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/9zgdmr0rir5ljq5il6pc7qfps5m2s5xb-python3.10-jupyterhub-ldapauthenticator-1.3.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/y1bhw1nhrz47i9jg238myzambqd3mpyy-python3.10-jupyterhub-systemdspawner-0.15.drv' failed to build
error: 1 dependencies of derivation '/nix/store/s6rbl0c9ax3bxhzdh5m62bxmczky136q-python3.10-jupyterhub-tmpauthenticator-0.6.drv' failed to build
error: 1 dependencies of derivation '/nix/store/hv97n30pkymm0kd6jqnwhsm839p72970-python3.10-oauthenticator-14.2.0.drv' failed to build
error: 22 dependencies of derivation '/nix/store/357vyazmi8fmpi87jayq2sacrq8h54q7-env.drv' failed to build
error: 1 dependencies of derivation '/nix/store/787a642qnqf4fqjjbqrzil8pv8myf8nh-review-shell.drv' failed to build
```

If no one objects I'll go ahead and merge tomorrow.

--- a/pkgs/development/python-modules/nbclient/default.nix
+++ b/pkgs/development/python-modules/nbclient/default.nix
@@ -1,16 +1,18 @@
-{ lib
+{ async_generator
 , buildPythonPackage
-, fetchPypi
-, pythonOlder
-, async_generator
-, traitlets
+, fetchFromGitHub
+, ipykernel
+, ipywidgets
+, jupyter-client
+, lib
+, nbconvert
 , nbformat
 , nest-asyncio
-, jupyter-client
+, pytest-asyncio
 , pytestCheckHook
+, pythonOlder
+, traitlets
 , xmltodict
-, nbconvert
-, ipywidgets
 }:
 
 let nbclient = buildPythonPackage rec {
@@ -20,9 +22,11 @@ let nbclient = buildPythonPackage rec {
 
   disabled = pythonOlder "3.7";
 
-  src = fetchPypi {
-    inherit pname version;
-    hash = "sha256-uAcm/B+4mg6Pi+HnfijQAmsejtkLwUPIoMdiLk+M3Z4=";
+  src = fetchFromGitHub {
+    owner = "jupyter";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-ofyJNJeNkXbZ9qHLTuzJ13PgEXU9lDQ9NkT4nG/R8Ic=";
   };
 
   propagatedBuildInputs = [ async_generator traitlets nbformat nest-asyncio jupyter-client ];
@@ -30,7 +34,14 @@ let nbclient = buildPythonPackage rec {
   # circular dependencies if enabled by default
   doCheck = false;
 
-  checkInputs = [ pytestCheckHook xmltodict nbconvert ipywidgets ];
+  checkInputs = [
+    ipykernel
+    ipywidgets
+    nbconvert
+    pytest-asyncio
+    pytestCheckHook
+    xmltodict
+  ];
 
   preCheck = ''
     export HOME=$(mktemp -d)

